sdk/src/azure/core/az_json_builder.c

```diff
@@ -494,6 +513,46 @@ az_json_builder_append_int32_number(az_json_builder* json_builder, int32_t value
   return AZ_OK;
 }
 
+AZ_NODISCARD az_result az_json_builder_append_double_number(
```

Comment:
shorten name to az_json_builder_append_double or maybe az_json_builder_append_number?

Reply:
I am OK with:
`az_json_builder_append_double`
`az_json_builder_append_int32`

`az_json_builder_append_number` won't work because we can overload it for the different numeric types.

The reason for the long name, that has the `number` prefix was to be consistent with the other overloads that end with the type of JSON primitive we are writing, for example:

Let me know.